src/mds/Server.cc

```diff
@@ -1291,6 +1204,100 @@ void Server::respond_to_request(MDRequestRef& mdr, int r)
   }
 }
 
+// statistics mds req op number and latency 
+void Server::perf_gather_op_latency(__u32 op, utime_t lat)
```

Comment:
Have this take the client_request as the first argument instead of the op #.

Reply:
Is there any benefit in doing this? The function perf_gather_op_latency only use the op num, if we pass the req will increase once more copy of structure  MClientRequest.

Reply:
> Is there any benefit in doing this?

You don't need to codify the type of `op` and other members may be used in the future.

> The function perf_gather_op_latency only use the op num, if we pass the req will increase once more copy of structure MClientRequest.

Not if you use `MClientRequest &req`.

Reply:
okay